src/Tasks/Microsoft.NET.Build.Tasks/GetDependsOnNETStandard.net46.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if NET46
+using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public partial class GetDependsOnNETStandard
+    {
+        private static Guid s_importerGuid = new Guid("7DAC8207-D3AE-4c75-9B67-92801A497D44");
+
+        internal static bool GetFileDependsOnNETStandard(string filePath)
+        {
+            // Ported from Microsoft.Build.Tasks.AssemblyInformation
+            if (Environment.OSVersion.Platform == PlatformID.Unix || Environment.OSVersion.Platform == PlatformID.MacOSX)
```

Comment:
I tested it manually in debugger.  I could provide a hook for testing it on Windows.